AfterClass-io/afterclass.io-v2
# feat(devtool): local development environment with docker

I realise this PR might be a little huge, let me know if it'll be better if I were to do a separate one to address cleanups & deps version bumps instead

## Changes

### core

- mock local supabase development environment with docker
- mock data at `prisma/data/*.json`
- db seeding logic at `prisma/seed.ts`
- updated db schema to be less ambiguous and support relevant data required for reviews
- update `.env.example` with working local dev db url

### optional

- deps version bump
  - breaking changes to trpc (low impact due to lack of features using trpc)
  - minor changes to `/account/auth/login` & `/account/auth/verify` pages, wrap `<Suspense>` around components using `useSearchParam` hook
- cleanup unused `.vscode/settings.json` fields
- add `.vscode/extensions.json` recommendations

## Testing

1. ensure that docker compose is installed
   ```sh
   docker compose version
   ```
   or
   ```sh
   docker-compose --version
   ```
4. run docker compose to spin up db container
   ```sh
   docker compose up -d
   ```
5. make a copy of `.env.example` and rename it to `.env`
6. run db migration + seeding command
   ```sh
   npx prisma migrate dev
   npx prisma db seed
   ```
   or
   ```sh
   yarn db:reset
   ```
7. confirm with a db client that database is populated as expected
   ![image](https://github.com/AfterClass-io/afterclass.io-v2/assets/13061926/0cd4829d-4fd8-495d-8171-50ab6fc3d2fc)
8. run `yarn dev`
9. navigate to `/` after logging in as `test@test.com` user
10. confirm that data is present
    ![image](https://github.com/AfterClass-io/afterclass.io-v2/assets/13061926/7287ad5c-f6fc-4dde-8705-98adf9b3fe54)
11. changing any values from `public.university` should reflect on `/` upon refresh
